### PR TITLE
ReactiveXC-5

### PR DESCRIPTION
**Pull request by yahyanajar**
Unit tests for rabbitMq

### CODE REVIEW

**Comment by Julien-Molina** [comment]
test this please

**Comment by nicolaserny** [comment]
Can one of the admins verify this patch?

**Comment by Julien-Molina** [comment]
retest this please

**Comment by nicolaserny** [comment]
Can one of the admins verify this patch?

**Comment by nicolaserny** [comment]
test this please

**Comment by nicolaserny** [comment]
retest this please

**Comment by nicolaserny** [comment]
ok to test

**Comment by nicolaserny** [comment]
retest this please

File: ReactiveXComponent/XCApi/XComponentApi.cs [hunk 1]
```diff
@@ -1,135 +1,49 @@
...
- 
-            var publisher = this._publisherFactory.Create(component);
-            publisher.Send(header, message, this._parser.GetPublisherTopic(component, stateMachine, eventCode));
-            publisher.Close();
+            _xcConnection = new XCConnection(connection);
+            _xcSessionFactory = new XCSessionFactory(publisherFactory, consumerFactory, parser);
         }
 
-        public void AddCallback(string component, string stateMachine, Action<MessageEventArgs> callback)
+        public void CreateSession()
```
Comment (nicolaserny) [inline ReactiveXComponent/XCApi/XComponentApi.cs]: you should return your session ?

File: ReactiveXComponent/XCApi/XComponentApi.cs [hunk 2]
```diff
@@ -1,135 +1,49 @@
...
-                        {
-                            callBack(args);
-                        }
-                    }
-                }
-                
-            };
-            _privateConsumer.Start();
-
+            _xcConnection.Close();
```
Comment (nicolaserny) [inline ReactiveXComponent/XCApi/XComponentApi.cs]: closing connection is not enough ?

**Comment by nicolaserny** [comment]
@yahyanajar @Julien-Molina  Perhaps it could be easier and quicker to start again from a clean pullrequest with only a small feature such as connection and session. And add features in next pull requests.

File: ReactiveXComponent/XCApi/XComponentApi.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public class XComponentApi : IXComponentApi
+    {
+        private bool _disposed = false;
+        
+        private readonly XCConnection _xcConnection;
+        private readonly XCSessionFactory _xcSessionFactory;
+        private XCSession _xcSession;
+
+        public XComponentApi(IRabbitMqPublisherFactory publisherFactory, IRabbitMqConsumerFactory consumerFactory, IRabbitMqConnection connection, DeploymentParser parser)
```
Comment (nicolaserny) [inline ReactiveXComponent/XCApi/XComponentApi.cs]: strange constructor !
do you think our lib user will provide such parameters ??

File: ReactiveXComponent/XCApi/XComponentApi.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            _xcConnection = new XCConnection(connection);
```
Comment (nicolaserny) [inline ReactiveXComponent/XCApi/XComponentApi.cs]: you never use your connection !! strange
your connection should create a session...

**Pull request closed without merging**